src/turn_integrator.py

```diff
@@ -25,22 +26,42 @@ def turn_credentials(line_name):
 
     return eval_credentials(config_json)
 
-### Obtain a new auth token from Turn.io
-### TODO Not tested yet
-def obtain_auth_token(username, password):
-    u_pass = HTTPBasicAuth(username, password)
-    return requests.post('https://whatsapp.turn.io/v1/users/login', auth=u_pass)
+### CONTACTS
+## Obtain a contact profile. See documentation here:
+## https://whatsapp.turn.io/docs/api/contacts#retrieve-a-contact-profile
+def obtain_contact_profile(msisdn: str, line_name: str) -> requests.Response:
+    auth_headers = {
+        'Authorization': f'Bearer {turn_credentials(line_name)}',
+        'Accept': 'application/vnd.v1+json'
+    }
+
+    response = requests.get(f'https://whatsapp.turn.io/v1/contacts/{msisdn}/profile', headers=auth_headers)
+    print(response.text)
+    return response
+
+## Update a contact profile. Supply only the fields that need updating. See documentation here:
+## https://whatsapp.turn.io/docs/api/contacts#update-a-contact-profile
+def update_contact_profile(msisdn: str, line_name: str, profile_data: json) -> requests.Response:
+    auth_headers = {
+        'Authorization': f'Bearer {turn_credentials(line_name)}',
+        'Accept': 'application/vnd.v1+json'
+    }
+
+    response = requests.patch(f'https://whatsapp.turn.io/v1/contacts/{msisdn}/profile', headers=auth_headers, json=profile_data)
+    print(response.text)
+    return response
 
-### Send the different kinds of messages. See documentation here:
-### https://whatsapp.turn.io/docs/api/messages
-def send_message(message_data, line_name):
+### MESSAGES
+## Send the different kinds of messages. See documentation here:
+## https://whatsapp.turn.io/docs/api/messages
+def send_message(line_name: str, message_data: str) -> requests.Response:
     auth_headers = {
         'Authorization': f'Bearer {turn_credentials(line_name)}'
     }
 
     return requests.post('https://whatsapp.turn.io/v1/messages', headers=auth_headers, json=message_data)
 
-def send_text_message(msisdn, message, line_name):
+def send_text_message(msisdn: str, line_name: str, message: str) -> requests.Response:
     message_data = {
         'preview_url': False,
         'recipient_type': 'individual',
```

Comment:
I don't see anything in the turn documentation for other recipient_type values, so I think this is fine, but probably worth noting that it is sent as a hard coded value in the docstrings for this function.

Reply:
Doc info added.